dev-docs/source/Testing/SANSGUI/SANSGUITests.rst

```diff
@@ -0,0 +1,76 @@
+.. _sans_gui_testing:
+
+SANS GUI Testing
+================
+
+.. contents::
+   :local:
+
+Data reduction
+--------------
+
+*Preparation*
+
+-  Get the ISIS sample data from the website
+-  Ensure that the ISIS sample data directory is in the search directories
+
+
+**Time required 20 - 30 minutes**
+
+--------------
+
+#. Open ``Interfaces`` > ``SANS`` > ``SANS v2``
+#. Choose ``Load User File``; from the ISIS sample data in the ``loqdemo`` folder, choose ``Maskfile.txt``
+#. Choose ``Load Batch File``; from the ISIS sample data in the `loqdemo` folder, choose ``batch_mode_reduction.csv``
+#. In the ``Settings`` tab:
+    - Go to ``Mask``
+    - Click ``Display Mask``
+    - This should give an instrument view with a circle at the centre
+#. In the ``Runs`` tab
+	- Click ``Process All``
+	- After some seconds the rows should turn green
+	- In the Main window there should be a series of new workspaces; 3 group workspaces and 4 2D workspaces
+	- Change the first column of the first row to 74045; click process
+	- The row should turn blue; hovering over the row should give an error message
+	- Change the first column of the first row back to 74044
+	- Change the ``Reduction`` button to 2D
+	- Check the ``Plot Results`` box
+	- Click ``Process All``
+	- A plot window will open; initially empty, then with a red line
+	- Change ``Reduction`` back to 1D
+	- Click ``Process All``
+	- In the plot you should end up with a red and a black line plotted
+	- Check the ``Multi-period`` box
+	- 6 additional rows should appear in the table
+	- Check that the ``Add``, ``Delete``, ``Copy``, ``Paste``, ``Cut`` and ``Erase`` icons work as expected
+	- Delete all rows and re-load the batch file as previously
+#. In the ``Settings`` tab
+	- Set ``Reduction Type`` to ``Merged``
+	- Return to the ``Runs`` tab
+	- Ensure ``Plot results`` is checked
+	- Click ``Process All``
+	- This should result in a plot with three lines
+#. In the ``Beam centre`` tab
+	- Click Run
+	- A plot should appear after some seconds, with 4 lines
+	- The 4 lines should gradually get closer together
+	- This will run for some time, probably minutes
+#. In the ``Sum Runs`` tab
+		- Enter ``74044, 74019`` in the top line
+		- Enter ``LOQ74044-add`` as Output file
```

Comment:
Need to click `add` before clicking `sum`. Once you click add the Output file name will be populated